src/math/Color.js

```diff
@@ -62,39 +62,54 @@ THREE.Color.prototype = {
 
 	},
 
-	setHSL: function ( h, s, l ) {
+	setHSL: function() {
+		// comupute euclidian module of x % n
+		function emod ( x, n ) {
+			return x >= 0 ? ( x % n ) : ( ( n - ( -x % n ) ) % n );
```

Comment:
I believe the common fix for this is:

    function mod2( n, m ) {

        return ( ( n % m ) + m ) % m;

    }